src/jaxsim/simulation/integrators.py

```diff
@@ -526,94 +437,16 @@ def body_fun(carry: Tuple, idx: int) -> Tuple[Tuple, jtp.PyTree]:
-) -> Union[State, Tuple[State, Dict[str, Any]]]:
-    """
-    Integrate a system of ODEs using the Euler method.
-
-    Args:
-        func: A function that computes the time-derivative of the state.
-        y0: The initial state.
-        t: The vector of time instants of the integration horizon.
-        *args: Additional arguments to be passed to the function func.
-        num_sub_steps: The number of sub-steps to be performed within each integration step.
-        return_aux: Whether to return the auxiliary data produced by the integrator.
-
-    Returns:
-        The state of the system at the end of the integration horizon, and optionally
-        the auxiliary data produced by the integrator.
-    """
-
-    # Close func over additional inputs and parameters
-    dx_dt_closure_aux = lambda x, ts: func(x, ts, *args)
-
-    # Close one-step integration over its arguments
-    integrator_single_step = lambda t0, tf, x0: odeint_euler_one_step(
-        dx_dt=dx_dt_closure_aux, x0=x0, t0=t0, tf=tf, num_sub_steps=num_sub_steps
-    )
-
-    # Integrate the state and compute optional auxiliary data over the horizon
-    out, aux = integrate_single_step_over_horizon(
-        integrator_single_step=integrator_single_step, t=t, x0=y0
-    )
-
-    return (out, aux) if return_aux else out
-
-
-def odeint_euler_semi_implicit(
-    func,
-    y0: ODEState,
-    t: TimeHorizon,
-    *args,
-    num_sub_steps: int = 1,
-    return_aux: bool = False
-) -> Union[ODEState, Tuple[ODEState, Dict[str, Any]]]:
-    """
-    Integrate a system of ODEs using the Semi-Implicit Euler method.
-
-    Args:
-        func: A function that computes the time-derivative of the state.
-        y0: The initial state as ODEState object.
-        t: The vector of time instants of the integration horizon.
-        *args: Additional arguments to be passed to the function func.
-        num_sub_steps: The number of sub-steps to be performed within each integration step.
-        return_aux: Whether to return the auxiliary data produced by the integrator.
-
-    Returns:
-        The state of the system at the end of the integration horizon as ODEState object,
-        and optionally the auxiliary data produced by the integrator.
-    """
-
-    # Close func over additional inputs and parameters
-    dx_dt_closure_aux = lambda x, ts: func(x, ts, *args)
-
-    # Close one-step integration over its arguments
-    integrator_single_step = lambda t0, tf, x0: odeint_euler_semi_implicit_one_step(
-        dx_dt=dx_dt_closure_aux, x0=x0, t0=t0, tf=tf, num_sub_steps=num_sub_steps
-    )
-
-    # Integrate the state and compute optional auxiliary data over the horizon
-    out, aux = integrate_single_step_over_horizon(
-        integrator_single_step=integrator_single_step, t=t, x0=y0
-    )
-
-    return (out, aux) if return_aux else out
-
-
-def odeint_rk4(
+def odeint(
     func,
     y0: State,
     t: TimeHorizon,
     *args,
+    method: str = "euler",
```

Comment:
Please use the enum also here

Reply:
Thanks, done in https://github.com/ami-iit/jaxsim/pull/82/commits/ead3fb3098ff580f05ef05e76cbe6280913dd9f3